File: zantt-api-backend/src/main/java/com/zantt/services/ProjectService.java
package com.zantt.services;

import com.zantt.entities.ProjectEntity;
import com.zantt.repositories.ProjectRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public class ProjectService {
    @Autowired
    ProjectRepository projectRepository;

    public List<ProjectEntity> getProjects() {
        return projectRepository.findAll();
    }

    public ProjectEntity getProject(String projectId) {
        var project = projectRepository.findById(projectId);
        if (project.isEmpty()) {
            return null;
        }
        return project.get();
    }

    public ProjectEntity addProject(String name) {
        var projectId = UUID.randomUUID().toString();

        var project = new ProjectEntity();
        project.setProjectId(projectId);
        project.setName(name);

        return projectRepository.save(project);
    }

    public void deleteProject(String projectId) {
        projectRepository.deleteById(projectId);
    }
}

File: zantt-api-backend-net/Zantt/Services/ProjectService.cs
using System.Diagnostics;
using Zantt.Entities;
using Zantt.Exceptions;
using Zantt.Repositories;

namespace Zantt.Services;

public class ProjectService
{
    private readonly ProjectRepository projectRepository;
    private readonly ILogger<ProjectService> logger;

    public ProjectService(
        ILogger<ProjectService> logger,
        ProjectRepository projectRepository)
    {
        this.logger = logger;
        this.projectRepository = projectRepository;
    }

    public List<ProjectEntity> GetProjects()
    {
        return projectRepository.GetProjects();
    }

    public ProjectEntity? GetProject(string projectId)
    {
        if (projectId == null)
        {
            throw new WellKnownApiException($"{nameof(projectId)} is null", "INVALID_PARAMETER");
        }

        return projectRepository.GetProjectByProjectId(projectId);
    }

    public ProjectEntity? AddProject(string name)
    {
        if (name == null)
        {
            throw new WellKnownApiException($"{nameof(name)} is null", "INVALID_PARAMETER");
        }

        return projectRepository.AddProject(name);
    }

    public ProjectEntity? UpdateProject(string projectId, string name)
    {
        if (projectId == null)
        {
            throw new WellKnownApiException($"{nameof(projectId)} is null", "INVALID_PARAMETER");
        }
        if (name == null)
        {
            throw new WellKnownApiException($"{nameof(name)} is null", "INVALID_PARAMETER");
        }

        return projectRepository.UpdateProjectByProjectId(projectId, name);
    }

    public void DeleteProject(string projectId)
    {
        if (projectId == null)
        {
            throw new WellKnownApiException($"{nameof(projectId)} is null", "INVALID_PARAM");
        }

        projectRepository.DeleteProjectByProjectId(projectId);
    }
}
